Index: orca/src/orca/orca.in
===================================================================
--- orca/src/orca/orca.in	(revision 3817)
+++ orca/src/orca/orca.in	(working copy)
@@ -272,9 +272,13 @@ else
         #
         if [ `egrep -c "\-s|\-g|\-t|\-n|\-u|\-e|\-d" <<< $ARGS` -eq 0 ]
         then
-            runOrca "NO_CLEANUP"
+            runOrca "NO_CLEANUP" &
+            orca_pid=$!
+            wait $orca_pid
         else
-            runOrca
+            runOrca &
+            orca_pid=$!
+            wait $orca_pid
         fi
     fi
 fi
